mitosheet/mitosheet/step_performers/graph_steps/graph_rename.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+from copy import copy
+from typing import Any, Dict, List, Optional, Set, Tuple
+
+from mitosheet.state import State
+from mitosheet.step_performers.step_performer import StepPerformer
+from mitosheet.types import GraphID
+from mitosheet.utils import get_valid_dataframe_name
+
+
+class GraphRenameStepPerformer(StepPerformer):
+    """"
+    A rename graph step changes the name of a specific graph id
+    """
+
+    @classmethod
+    def step_version(cls) -> int:
+        return 1
+
+    @classmethod
+    def step_type(cls) -> str:
+        return 'graph_rename'
+
+    @classmethod
+    def step_display_name(cls) -> str:
+        return 'Renamed a Graph'
+    
+    @classmethod
+    def step_event_type(cls) -> str:
+        return 'graph_rename_edit'
+
+    @classmethod
+    def saturate(cls, prev_state: State, params: Dict[str, Any]) -> Dict[str, Any]:
+        graph_id = params['graph_id']
+        old_graph_tab_name = prev_state.graph_data[graph_id]["graphTabName"]
+        params['old_graph_tab_name'] = old_graph_tab_name
+        return params
+
+    @classmethod
+    def execute( # type: ignore
+        cls,
+        prev_state: State,
+        graph_id: GraphID,
+        old_graph_tab_name: str,
+        new_graph_tab_name: str,
+        **params
+    ) -> Tuple[State, Optional[Dict[str, Any]]]:
+        # Bail early, if there is no change or the new name is empty
+        if old_graph_tab_name == new_graph_tab_name or new_graph_tab_name == '' :
+            return prev_state, None
+
+        # Create a new step and save the parameters
+        post_state = copy(prev_state)
+
+        post_state.graph_data[graph_id]["graphTabName"] = new_graph_tab_name
```

Comment:
@aarondr77 I think we should always make the graphTabName be the same as the name of the graph itself; like, the name of the graph is the name of the variable of the graph is the name of the graph itself (with underscores). 

Not that I think we should do that right now. I _do think_ that this is something that we should specify and explore, as I actually think there's a bit of design work that we can do!

Reply:
We can add this specification to the other specification if you want, but I think we should lead it out of this PR -- feature creep is real, careful!